third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -6,233 +6,154 @@
 #include <realdds/dds-participant.h>
 #include <realdds/dds-publisher.h>
 #include <realdds/dds-topic-writer.h>
+#include <realdds/dds-topic.h>
 #include <realdds/dds-utilities.h>
-#include <realdds/dds-guid.h>
 #include <realdds/topics/dds-topic-names.h>
 #include <realdds/topics/flexible/flexible-msg.h>
-#include <realdds/topics/flexible/flexiblePubSubTypes.h>
 
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/core/condition/WaitSet.hpp>
+#include <fastdds/dds/core/condition/GuardCondition.hpp>
+
+#include <rsutils/shared-ptr-singleton.h>
 #include <rsutils/string/slice.h>
 using rsutils::string::slice;
 
-#include <iostream>
 
-using namespace eprosima::fastdds::dds;
-using namespace realdds;
+namespace realdds {
+
 
-// We want to know when readers join our topic
-class dds_device_broadcaster::dds_client_listener : public realdds::dds_topic_writer //dds_topic_writer is a listener
+// Singleton, per participant
+// Manages the thread from which broadcast messages are sent
+//
+class detail::broadcast_manager
 {
+    std::thread _th;
+    std::shared_ptr< dds_topic_writer > _writer;
+    eprosima::fastdds::dds::GuardCondition _stopped, _ready_for_broadcast;
```

Comment:
Please separate to 2 lines. It is clearer there are 2 member variables like that.

Reply:
Done